SIPS/sip-78.md

```diff
@@ -0,0 +1,77 @@
+
+[SIP-65](./sip-65.md) introduced a decentralized circuit breaker which is tripped when a price on-chain is detected more than some given factor. When an iSynth is reset with a new `entryPoint` however, this changes the price quite significantly, and can trip the circuit breaker when there is a previous `lastExchangeRate` in `Exchanger` for that synth. This makes the synth unusable unless the factor is bumped up sufficiently to handle the iSynth repricing (or any iSynth repricing if many are performed simultaneously), and since the factor setting is system-wide, a much higher factor makes SIP-65 less useful overall.
+
+## Specification
+
+<!--The specification should describe the syntax and semantics of any new feature, there are five sections
+1. Overview
+2. Rationale
+3. Technical Specification
+4. Test Cases
+5. Configurable Values
+-->
+
+### Overview
+
+<!--This is a high level overview of *how* the SIP will solve the problem. The overview should clearly describe how the new feature will be implemented.-->
+
+When the protocolDAO resets an iSynth, the `lastExchangeRate` in `Exchanger` must be updated to the current synth rate, to indicate a refreshed price for the synth.
+
+### Rationale
+
+<!--This is where you explain the reasoning behind how you propose to solve the problem. Why did you propose to implement the change in this way, what were the considerations and trade-offs. The rationale fleshes out what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+The protocolDAO already has the power to invoke `ExchangeRates.setInversePricing()`. It's at this point that the `Exchanger` must be told to update any previous exchange rate with the one calculated at the time of repricing.
+
+### Technical Specification
+
+<!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces Synthetix currently exposes or the creations of new ones.-->
+
+N/A
+
+### Test Cases
+
+<!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
+
+- Given `iETH` is an inverse synth tracking `ETH`
+- And there exists a previous exchange into or out of `iETH`, ensuring `Exchanger.lastExchangeRate(iETH)` has an entry
+- When the protocolDAO invokes setInversePricing, changing the rate more than the `Exchanger.priceDeviationThresholdFactor`
```

Comment:
Resetting / deleting `lastExchangeRate(iETH)` to 0 would technically work here too 

```
        uint lastRateFromExchange = lastExchangeRate[currencyKey];

        if (lastRateFromExchange > 0) {
            return _isDeviationAboveThreshold(lastRateFromExchange, currentRate);
        }
```

Reply:
Actually it would instead then look up the last 3 prices on-chain, which after SIP-75 in Pollux, would mean calculating the iSynth from previous long rates. This would still work in most cases, but isn't as accurate as setting an explicit `lastRateFromExchange` for the iSynth

Reply:
@justinjmoses  Yeah , should be setting it to the new `EntryPoint` of the iSynths that it's getting resetted to. So the system will acknowledge last trade price is new 'EntryPoint'

Reply:
well it will set it to the new iSynth price calculated from the current long rate and the new entryPoint. That's what you meant right?